lib/actions/app.ts

```diff
@@ -0,0 +1,227 @@
+	permission: 'user',
+	primary: true,
+	async action(_params, _options, done) {
+		const _ = await import('lodash');
+		const balena = (await import('balena-sdk')).fromSharedOptions();
+		const visuals = await import('resin-cli-visuals');
+
+		return balena.models.application
+			.getAll({
+				$select: ['id', 'app_name', 'device_type'],
+				$expand: { owns__device: { $select: 'is_online' } },
+			})
+			.then(
+				(
+					applications: Array<
+						Application & { device_count?: number; online_devices?: number }
+					>,
+				) => {
+					applications.forEach(application => {
+						application.device_count = _.size(application.owns__device);
+						application.online_devices = _.sumBy(application.owns__device, d =>
+							d.is_online === true ? 1 : 0,
+						);
+					});
+
+					console.log(
+						visuals.table.horizontal(applications, [
+							'id',
+							'app_name',
+							'device_type',
+							'online_devices',
+							'device_count',
+						]),
+					);
+				},
+			)
+			.nodeify(done);
+	},
+};
+
+export const info: CommandDefinition<{
+	name: string;
+}> = {
+	signature: 'app <name>',
+	description: 'list a single application',
+	help: `\
+Use this command to show detailed information for a single application.
+
+Examples:
+
+	$ balena app MyApp\
+`,
+	permission: 'user',
+	primary: true,
+	async action(params, _options, done) {
+		const balena = (await import('balena-sdk')).fromSharedOptions();
+		const visuals = await import('resin-cli-visuals');
+
+		return balena.models.application
+			.get(params.name)
+			.then(application => {
+				console.log(
+					visuals.table.vertical(application, [
+						`$${application.app_name}$`,
+						'id',
+						'device_type',
+						'slug',
+						'commit',
+					]),
+				);
+			})
+			.nodeify(done);
+	},
+};
+
+export const restart: CommandDefinition<{
+	name: string;
+}> = {
+	signature: 'app restart <name>',
+	description: 'restart an application',
+	help: `\
+Use this command to restart all devices that belongs to a certain application.
+
+Examples:
+
+	$ balena app restart MyApp\
+`,
+	permission: 'user',
+	async action(params, _options, done) {
+		const balena = (await import('balena-sdk')).fromSharedOptions();
+		return balena.models.application.restart(params.name).nodeify(done);
+	},
+};
+
+export const remove: CommandDefinition<
+	{ name: string },
+	commandOptions.YesOption
+> = {
+	signature: 'app rm <name>',
+	description: 'remove an application',
+	help: `\
+Use this command to remove a balena application.
+
+Notice this command asks for confirmation interactively.
+You can avoid this by passing the \`--yes\` boolean option.
+
+Examples:
+
+	$ balena app rm MyApp
+	$ balena app rm MyApp --yes\
+`,
+	options: [commandOptions.yes],
+	permission: 'user',
+	async action(params, options, done) {
+		const balena = (await import('balena-sdk')).fromSharedOptions();
+		const patterns = await import('../utils/patterns');
+
+		return patterns
+			.confirm(
+				options.yes ?? false,
```

Comment:
ooh [Nullish Coalescing](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing)! :-) I think it's the first usage in the CLI repo. In this case, though, I think `!!options.yes` would have worked as well.